File: code examples/golang/amazon_search.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
)

func main() {
	const Username = "YOUR_USERNAME"
	const Password = "YOUR_PASSWORD"

	payload := map[string]interface{}{
		"source":     "amazon_search",
		"domain":     "nl",
		"query":      "adidas",
		"start_page": 11,
		"pages":      10,
		"parse":      true,
		"context": []map[string]interface{}{
			{"key": "category_id", "value": 16391843031},
			{"key": "merchant_id", "value": "3AA17D2BRD4YMT0X"},
		},
	}

	jsonValue, _ := json.Marshal(payload)

	client := &http.Client{}
	request, _ := http.NewRequest("POST",
		"https://realtime.oxylabs.io/v1/queries",
		bytes.NewBuffer(jsonValue),
	)

	request.SetBasicAuth(Username, Password)
	response, _ := client.Do(request)

	responseText, _ := ioutil.ReadAll(response.Body)
	fmt.Println(string(responseText))
}

File: code examples/php/amazon_search.php
<?php

$params = array(
    'source' => 'amazon_search',
    'domain' => 'nl',
    'query' => 'adidas',
    'start_page' => 11,
    'pages' => 10,
    'parse' => true,
    'context' => [
      [
        'key' => 'category_id', 
        'value' => 16391843031,
      ],
      [
        'key' => 'merchant_id',
        'value' => '3AA17D2BRD4YMT0X'
      ]
    ],
);

$ch = curl_init();

curl_setopt($ch, CURLOPT_URL, "https://realtime.oxylabs.io/v1/queries");
curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($params));
curl_setopt($ch, CURLOPT_POST, 1);
curl_setopt($ch, CURLOPT_USERPWD, "user" . ":" . "pass1");


$headers = array();
$headers[] = "Content-Type: application/json";
curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);

$result = curl_exec($ch);
echo $result;

if (curl_errno($ch)) {
    echo 'Error:' . curl_error($ch);
}
curl_close ($ch);
?>